src/SignalR/common/Http.Connections/test/MapConnectionHandlerTests.cs

```diff
@@ -34,39 +34,80 @@ public MapConnectionHandlerTests(ITestOutputHelper output)
         public void MapConnectionHandlerFindsAuthAttributeOnEndPoint()
         {
             var authCount = 0;
-            using (var builder = BuildWebHost<AuthConnectionHandler>("/auth",
+            using (var host = BuildWebHost<AuthConnectionHandler>("/auth",
                 options => authCount += options.AuthorizationData.Count))
             {
-                builder.Start();
+                host.Start();
+
+                var dataSource = host.Services.GetRequiredService<EndpointDataSource>();
+                // We register 2 endpoints (/negotiate and /)
+                Assert.Equal(2, dataSource.Endpoints.Count);
+                Assert.Equal(1, dataSource.Endpoints[0].Metadata.GetOrderedMetadata<IAuthorizeData>().Count);
+                Assert.Equal(1, dataSource.Endpoints[1].Metadata.GetOrderedMetadata<IAuthorizeData>().Count);
             }
 
-            Assert.Equal(1, authCount);
+            Assert.Equal(0, authCount);
         }
 
         [Fact]
         public void MapConnectionHandlerFindsAuthAttributeOnInheritedEndPoint()
         {
             var authCount = 0;
-            using (var builder = BuildWebHost<InheritedAuthConnectionHandler>("/auth",
+            using (var host = BuildWebHost<InheritedAuthConnectionHandler>("/auth",
                 options => authCount += options.AuthorizationData.Count))
             {
-                builder.Start();
+                host.Start();
+
+                var dataSource = host.Services.GetRequiredService<EndpointDataSource>();
+                // We register 2 endpoints (/negotiate and /)
+                Assert.Equal(2, dataSource.Endpoints.Count);
+                Assert.Equal(1, dataSource.Endpoints[0].Metadata.GetOrderedMetadata<IAuthorizeData>().Count);
+                Assert.Equal(1, dataSource.Endpoints[1].Metadata.GetOrderedMetadata<IAuthorizeData>().Count);
```

Comment:
You can use `Assert.Collection` for this (with a sort) and verify the details that are in the comments. It's a little more verbose, but also a lot more precise.

Reply:
I wasn't a fan when I first saw this, but after writing some, I kind of like it.